platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/turtle/TurtleNode.java

```diff
@@ -293,21 +295,33 @@ private void checkLifeCycle(@NonNull final LifeCycle expected, @NonNull final St
 
     private void doShutdownNode() throws InterruptedException {
         if (lifeCycle == LifeCycle.STARTED) {
-            // TODO: Release all resources
             getMetricsProvider().removePlatformMetrics(platform.getSelfId());
-            platformWiring.stop();
-            platform.getNotificationEngine().unregisterAll();
+            platform.destroy();
             platformStatus = null;
             platform = null;
             platformWiring = null;
             model = null;
+
+            final int childrenCount = savedTurtleAppState.getNumberOfChildren();
+            for (int i = 0; i < childrenCount; i++) {
```

Comment:
This must be wrong. We should not have to iterate over the children in a tree. We should not even know that there is a tree.

Reply:
We should free the resources of the tree, mainly the pipeline. This is one of the places which accumulates a lot of threads usage. Do you have an idea how to do it differently?

Reply:
I extracted this logic inside `TurtleAppState`